packages/react/src/ReactElement.js

```diff
@@ -341,6 +342,29 @@ export function createElement(type, config, children) {
     }
   }
 
+  if (__DEV__) {
+    if (warnAboutCreateElement) {
+      const displayName =
+        typeof type === 'function'
+          ? type.displayName || type.name || 'Unknown'
+          : type;
+      warningWithoutStack(
+        false,
+        '%s: You are most likely getting this warning because ' +
+          'you are spreading an object on an element while passing it a explicit ' +
+          'key in the format `<div {...props} key={keyVal} />`. To fix this, ' +
+          'switch the key and the props spread to `<div key={keyVal} {...props} />`.' +
+          '\n\n' +
+          'If you are explicitly calling React.createElement, please switch to ' +
+          'React.jsx instead.' +
```

Comment:
This is not correct. We don't want people to call `React.jsx` directly. That's an implementation detail for the JSX plugin. It's too easy to for example do stuff like `React.jsxs('div', { children: data.map(i => <div>Look ma, no keys</div>) })`.

If you're manually calling createElement, then that's fine. However, that probably means that we can't do the warning for the transform this way.